core/execution/future/market.go

```diff
@@ -2661,7 +2661,9 @@ func (m *Market) handleConfirmation(ctx context.Context, conf *types.OrderConfir
 
 		tradeEvts = append(tradeEvts, events.NewTradeEvent(ctx, *trade))
 		for _, mp := range m.position.Update(ctx, trade, conf.PassiveOrdersAffected[idx], conf.Order) {
-			m.marketActivityTracker.RecordPosition(m.settlementAsset, mp.Party(), m.mkt.ID, mp.Size(), trade.Price, m.positionFactor, m.timeService.GetTimeNow())
+			if twNotionalPosition := m.marketActivityTracker.RecordPosition(m.settlementAsset, mp.Party(), m.mkt.ID, mp.Size(), trade.Price, m.positionFactor, m.timeService.GetTimeNow()); twNotionalPosition != nil {
```

Comment:
this means that for an epoch with no position change there will be none and for epoch with many position changes there will be plenty. I think that's an overkill. Why not just emit an event at the end/beginning of the epoch with the tw notional position for all?

Reply:
Thanks, as advised I've added to the market activity tracker and published from there at the start of each epoch. The way I've done it, the rewards calculation will automatically calculate the time weighted notional positions anyway, so I've stored those values when they're calculated, and then at the start of the next epoch, I create the time weighted notional positions and set those for the epoch at the start of each epoch.